source/common/protobuf/utility.cc

```diff
@@ -48,7 +48,17 @@ ProtoValidationException::ProtoValidationException(const std::string& validation
   ENVOY_LOG_MISC(debug, "Proto validation error; throwing {}", what());
 }
 
+bool MessageUtil::allow_unknown_fields = false;
+
 void MessageUtil::loadFromJson(const std::string& json, Protobuf::Message& message) {
+  Protobuf::util::JsonParseOptions options;
+  const auto status = Protobuf::util::JsonStringToMessage(json, &message, options);
+  if (!status.ok()) {
+    throw EnvoyException("Unable to parse JSON as proto (" + status.ToString() + "): " + json);
+  }
+}
+
+void MessageUtil::loadFromJsonLenient(const std::string& json, Protobuf::Message& message) {
```

Comment:
Nit: could factor out some of the body of these two to a shared util.

Reply:
The other way of doing this is to have one function, and make the parameter for leniency be an enum instead of bool. That would be my vote as you would share code and have callsite clarity.

Reply:
Thanks. I combined the logic into one function which is only called from two places (json convert and a test). The old `loadFromJson` used in many places in tests, which makes me hesitant to change its signature.